pages/loserbox-stepper.js

```diff
@@ -0,0 +1,342 @@
+          setNetworkName('rinkeby')
+          setIsNetworkWarning(true)
+          break
+        case 5:
+          setNetworkName('goerli')
+          setIsNetworkWarning(true)
+          break
+        case 42:
+          setNetworkName('kovan')
+          break
+        default:
+          break
+      }
+
+      if (networkId == 1 || networkId == 42) {
+        let tokenContract
+        let mattContract
+        let decimals
+        let amount
+        if (networkId === 1) {
+          setEnvData({
+            ERCTOKEN: process.env.NEXT_PUBLIC_MAINNET_ERC_TOKEN,
+            MATTADDRESS:
+              process.env
+                .NEXT_PUBLIC_MAINNET_MULTIPLE_ARBITRABLE_CONTRACT_ADDRESS,
+            TIMEOUTPAYMENT: process.env.NEXT_PUBLIC_MAINNET_TIMEOUT_PAYMENT,
+            RECEIVER: process.env.NEXT_PUBLIC_MAINNET_RECEIVER_ADDRESS,
+            METAEVIDENCE: process.env.NEXT_PUBLIC_MAINNET_METAEVIDENCE,
+          })
+
+          tokenContract = new web3.eth.Contract(
+            erc20Abi,
+            process.env.NEXT_PUBLIC_MAINNET_ERC_TOKEN
+          )
+          mattContract = new web3.eth.Contract(
+            mattAbi,
+            process.env.NEXT_PUBLIC_MAINNET_MULTIPLE_ARBITRABLE_CONTRACT_ADDRESS
+          )
+          // Use BigNumber
+          decimals = web3.utils.toBN(
+            process.env.NEXT_PUBLIC_MAINNET_TOKEN_DECIMALS
+          )
+          amount = web3.utils.toBN(
+            process.env.NEXT_PUBLIC_MAINNET_TOKEN_AMOUNT
+          )
+        } else {
+          setEnvData({
+            ERCTOKEN: process.env.NEXT_PUBLIC_KOVAN_ERC_TOKEN,
+            MATTADDRESS:
+              process.env
+                .NEXT_PUBLIC_KOVAN_MULTIPLE_ARBITRABLE_CONTRACT_ADDRESS,
+            TIMEOUTPAYMENT: process.env.NEXT_PUBLIC_KOVAN_TIMEOUT_PAYMENT,
+            RECEIVER: process.env.NEXT_PUBLIC_KOVAN_RECEIVER_ADDRESS,
+            METAEVIDENCE: process.env.NEXT_PUBLIC_KOVAN_METAEVIDENCE,
+          })
+          tokenContract = new web3.eth.Contract(
+            erc20Abi,
+            process.env.NEXT_PUBLIC_KOVAN_ERC_TOKEN
+          )
+          mattContract = new web3.eth.Contract(
+            mattAbi,
+            process.env.NEXT_PUBLIC_KOVAN_MULTIPLE_ARBITRABLE_CONTRACT_ADDRESS
+          )
+          // Use BigNumber
+          decimals = web3.utils.toBN(
+            process.env.NEXT_PUBLIC_KOVAN_TOKEN_DECIMALS
+          )
+          amount = web3.utils.toBN(process.env.NEXT_PUBLIC_KOVAN_TOKEN_AMOUNT)
+        }
+
+        setTokenContract(tokenContract)
+
+        setMattContract(mattContract)
+        let value = amount.mul(web3.utils.toBN(10).pow(decimals))
+        setTokenAmount(value)
+
+        const userTokenBalance = await tokenContract.methods
+          .balanceOf(accounts[0])
+          .call()
+        if (userTokenBalance > value) {
+          setTokenBalanceApproved(true)
+        }
+        handleNext(networkId)
+      }
+    }
+  }
+
+  const handleNext = (networkId) => {
```

Comment:
It's more something like `handleNextStep` to make explicit it's connected with the stepper.